src/msg/async/AsyncMessenger.h

```diff
@@ -397,7 +397,7 @@ class AsyncMessenger : public SimplePolicyMessenger {
    */
   void unregister_conn(const AsyncConnectionRef& conn) {
     std::lock_guard l{deleted_lock};
-    if (!accepting_conns.count(conn))
+    if (!accepting_conns.count(conn) || (anon_conns.count(conn))
```

Comment:
@majianpeng  mind remove the parentheses around `anon_conns.count(conn)` ?

Reply:
@tchaikov . why remove?

Reply:
there is no need to put parentheses around this expression. why would you want to add them?

Reply:
in all in one ceph cluster, fio+librbd end up, the perfcounter(active_connections) didn't decrease.

Reply:
```suggestion
    if (!accepting_conns.count(conn) || anon_conns.count(conn)
```

is what i suggest.